python/ray/util/sgd/torch/examples/tune_example.py

```diff
@@ -47,15 +47,15 @@ def data_creator(config):
     return LinearDataset(2, 5), LinearDataset(2, 5, size=400)
 
 
-def tune_example(num_replicas=1, use_gpu=False):
+def tune_example(num_workers=1, use_gpu=False):
     config = {
         "model_creator": tune.function(model_creator),
         "data_creator": tune.function(data_creator),
         "optimizer_creator": tune.function(optimizer_creator),
         "loss_creator": tune.function(nn.MSELoss),
-        "num_replicas": num_replicas,
+        "num_workers": num_workers,
         "use_gpu": use_gpu,
-        "batch_size": 512,
+        "config": {"batch_size": 512 // num_workers},
```

Comment:
Stuff like this needs to be either very well documented, or the API needs to change to "total_batch_size" and automatically split per worker.

The reason that I am against having e.g. "batch_size_per_worker" btw is that in the latter case the training behavior will change with changes to `num_workers` since the affective batch size will be different. In the former case the training performance will change but not the training behavior, which imo is a much more sane default.

Reply:
Note that we right now try to maintain `config` purity - meaning that RaySGD does not perturb or change the given config dict in any way (it's basically in userland).

This PR, we move the `batch_size` parameter to userland, making it a hyperparameter. 

But I see your point; what if we have a documented `ray.util.sgd.BATCH_SIZE` constant that seamlessly handles scaling w/ number of workers?

Reply:
I think keeping `config` in userland is a good idea. Having a string constant is imo the best solution in the absence of better language tools if we want to keep `batch_size` in `config`. So yeah, sounds good!

Reply:
merged a fix - see `torch_trainer.py`. Just need to add docs.